test/integration/vaultauthmethods_integration_test.go

```diff
@@ -158,13 +158,33 @@ func TestVaultAuthMethods(t *testing.T) {
 				},
 			},
 		},
+		{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "vaultauth-test-approle",
+				Namespace: testK8sNamespace,
+			},
+			Spec: secretsv1alpha1.VaultAuthSpec{
+				// No VaultConnectionRef - using the default.
+				Namespace: testVaultNamespace,
+				Method:    "approle",
+				Mount:     "approle",
```

Comment:
I wonder if these values should either be TF vars, or outputs. That would prevent the test from becoming out of sync with the TF.